src/ray/common/task/task_spec.h

```diff
@@ -153,6 +153,8 @@ class TaskSpecification : public MessageWrapper<rpc::TaskSpec> {
 
   uint64_t ActorCounter() const;
 
+  void SetActorCounter(uint64_t actor_counter);
```

Comment:
I have mixed feelings about this change, because so far this class seems to only have immutable methods, but I need to update the sequence number before I retry submitting the task. An alternative would be to try to recreate the spec from scratch, but (1) it might be quite verbose and difficult to do from within the `TaskManager` and (2) it's likely to be slower.

Reply:
Yeah, I think it's okay to reset the actor counter. Normally we do try to keep the `TaskSpecification` const, but there are special cases where we do actually want to modify it. Another example of this is when we modify the task args to inline any small objects that are now available.